src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -140,10 +142,12 @@ export class TerminalInstance implements ITerminalInstance {
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
 
+		if (this._shellLaunchConfig && !this._shellLaunchConfig.name) {
```

Comment:
Let's move this into `_createProcess`, just before `this._messageTitleListener` is set. That way it's right next to the part that sets it for `_shellLaunchConfig.name`.